eth/downloader/downloader.go

```diff
@@ -304,6 +310,13 @@ func (d *Downloader) synchronise(id string, hash common.Hash, td *big.Int, mode
 		default:
 		}
 	}
+	for empty := false; !empty; {
```

Comment:
Though maybe we could have just added it to the above cycle which empties the wake channels.